### PR TITLE
Start using poetry and apply black

### PR DESCRIPTION
**Pull request by zsinx6**
- Move everything to pyproject.toml
- Update to use poetry, doing so we can have more simple way to have platform specific dependencies e.g. `pgi` (now it should work on every OS)
- Fixed Makefile
- Run CI for PRs using GitHub Actions

### CODE REVIEW

File: .github/workflows/test_push.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (julianaklulo) [inline .github/workflows/test_push.yaml:8]: Why is this needed?

Comment (zsinx6) [reply .github/workflows/test_push.yaml:8]: For the coveralls plugin, to be able to authenticate there.
This is an automatic token generated by github: https://docs.github.com/en/actions/security-guides/automatic-token-authentication

**Review by julianaklulo — APPROVED** [review]
:+1:
